src/Stratis.Bitcoin.Features.MemoryPool/Rules/CheckTxOutDustRule.cs

```diff
@@ -0,0 +1,38 @@
+﻿using Microsoft.Extensions.Logging;
+using NBitcoin;
+using NBitcoin.Policy;
+using Stratis.Bitcoin.Features.MemoryPool.Interfaces;
+
+namespace Stratis.Bitcoin.Features.MemoryPool.Rules
+{
+    /// <summary>
+    /// This rule ensures that the transaction does not contain any outputs which are considered dust.
+    /// </summary>
+    public sealed class CheckTxOutDustRule : MempoolRule
+    {
+        public CheckTxOutDustRule(
+            Network network,
+            ITxMempool mempool,
+            MempoolSettings mempoolSettings,
+            ChainIndexer chainIndexer,
+            ILoggerFactory loggerFactory) : base(network, mempool, mempoolSettings, chainIndexer, loggerFactory)
+        {
+        }
+
+        /// <inheritdoc />
+        public override void CheckTransaction(MempoolValidationContext context)
+        {
+            foreach (var txOut in context.Transaction.Outputs)
+            {
+                if (StandardTransactionPolicy.IsOpReturn(txOut.ScriptPubKey.ToBytes()))
+                    break;
```

Comment:
That should be continue. If we break than someone could exploit this and send 1000 dust payments with a single payment to nulldata address.

Reply:
Ahh well spotted 👍

Reply:
Ah yes of course 👍